bug: stdlib functions return float instead of preserving input type

## Description
Several standard library functions always return `float` even when operating on integer inputs, forcing users to cast the result or use float variables unexpectedly.

## Affected Functions
- `arrays.sum([int])` - returns float instead of int
- `arrays.min([int])` - returns float instead of int
- `arrays.max([int])` - returns float instead of int
- `math.abs(int)` - returns float instead of int
- `math.min(int, int)` - returns float instead of int
- `math.max(int, int)` - returns float instead of int

## Expected Behavior
Functions operating on integers should return integers. Functions operating on floats should return floats.

## Reproduction
```ez
import @std, @arrays, @math
using std

do main() {
    temp nums [int] = {1, 2, 3, 4, 5}
    temp sum int = arrays.sum(nums)  // Error: cannot assign float to int

    temp min_val int = math.min(5, 3)  // Error: cannot assign float to int
    temp abs_val int = math.abs(-5)    // Error: cannot assign float to int
}
```

## Actual Behavior
```
error[E3001]: type mismatch: cannot assign float to int
```

## Why This Is A Bug
- Type inconsistency: sum of integers should be an integer
- Requires unnecessary type casting or using float variables
- The comprehensive tests use `math.abs(-5)` directly in println, hiding this issue
- Common mathematical operations should preserve type when possible

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -2240,8 +2240,17 @@ func (tc *TypeChecker) inferStdCallType(funcName string, args []ast.Expression) 
 // inferMathCallType infers return types for @math functions
 func (tc *TypeChecker) inferMathCallType(funcName string, args []ast.Expression) (string, bool) {
 	switch funcName {
-	case "abs", "floor", "ceil", "round", "sqrt", "pow", "log", "log2", "log10",
-		"sin", "cos", "tan", "asin", "acos", "atan", "exp", "min", "max", "avg",
+	case "abs", "min", "max":
+		// Return type matches input type
+		if len(args) > 0 {
+			argType, ok := tc.inferExpressionType(args[0])
+			if ok && (argType == "int" || argType == "float") {
+				return argType, true
+			}
+		}
+		return "float", true // fallback
+	case "floor", "ceil", "round", "sqrt", "pow", "log", "log2", "log10",
+		"sin", "cos", "tan", "asin", "acos", "atan", "exp", "avg",
 		"random_float":
 		return "float", true
 	case "random", "factorial":
@@ -2273,8 +2282,19 @@ func (tc *TypeChecker) inferArraysCallType(funcName string, args []ast.Expressio
 		}
 		return "", false
 	case "sum", "product", "min", "max":
-		// Could be int or float depending on array type
-		return "float", true
+		// Return type matches array element type
+		if len(args) > 0 {
+			arrType, ok := tc.inferExpressionType(args[0])
+			if ok && len(arrType) > 2 && arrType[0] == '[' {
+				elemType := arrType[1 : len(arrType)-1]
+				// Handle fixed-size arrays like [int, 5]
+				if commaIdx := strings.Index(elemType, ","); commaIdx != -1 {
+					elemType = strings.TrimSpace(elemType[:commaIdx])
+				}
+				return elemType, true
+			}
+		}
+		return "float", true // fallback
 	case "avg":
 		return "float", true
 	case "reverse", "slice", "copy", "concat", "unique", "sorted", "filter", "map":

--- a/pkg/typechecker/typechecker_test.go
+++ b/pkg/typechecker/typechecker_test.go
@@ -1451,6 +1451,35 @@ do main() {
 	assertNoErrors(t, tc)
 }
 
+func TestArraysSumPreservesIntType(t *testing.T) {
+	input := `
+import @arrays
+
+do main() {
+	temp nums [int] = {1, 2, 3, 4, 5}
+	temp sum int = arrays.sum(nums)
+	temp min_val int = arrays.min(nums)
+	temp max_val int = arrays.max(nums)
+}
+`
+	tc := typecheck(t, input)
+	assertNoErrors(t, tc)
+}
+
+func TestMathAbsPreservesIntType(t *testing.T) {
+	input := `
+import @math
+
+do main() {
+	temp abs_val int = math.abs(-5)
+	temp min_val int = math.min(3, 7)
+	temp max_val int = math.max(3, 7)
+}
+`
+	tc := typecheck(t, input)
+	assertNoErrors(t, tc)
+}
+
 func TestStringsModuleCall(t *testing.T) {
 	input := `
 import @strings

